python_modules/libraries/dagster-dg/dagster_dg/cli/plus.py

```diff
@@ -144,3 +154,184 @@ def pull_env_command(**global_options: object) -> None:
+        global_ = True
+
+    if from_local_env:
+        local_env_value = (
+            ProjectEnvVars.from_ctx(dg_context).get(env_name) if dg_context.is_project else None
+        )
+        if local_env_value:
+            click.echo(f"Reading environment variable {env_name} from project .env file")
+        else:
+            local_env_value = os.getenv(env_name)
+            if local_env_value:
+                click.echo(f"Reading environment variable {env_name} from shell environment")
+            else:
+                raise click.UsageError(
+                    f"Environment variable {env_name} not found in your CLI environment or project .env file."
+                )
+        env_value = local_env_value
+
+    config = _get_config_or_error()
+
+    active_scopes = set(EnvVarScope(s) for s in scope) or {
+        EnvVarScope.FULL,
+        EnvVarScope.BRANCH,
+        EnvVarScope.LOCAL,
+    }
+    gql_client = DagsterCloudGraphQLClient.from_config(config)
+
+    location_suffix = "" if global_ else f" for location {dg_context.project_name}"
+    scope_text = f" in {', '.join(sorted(active_scopes))} scope"
+
+    existing_secrets = gql_client.execute(
+        gql.GET_SECRETS_FOR_SCOPES_QUERY,
+        variables={
+            "locationName": None if global_ else dg_context.project_name,
+            "scopes": {
+                "fullDeploymentScope": EnvVarScope.FULL in active_scopes,
+                "allBranchDeploymentsScope": EnvVarScope.BRANCH in active_scopes,
+                "localDeploymentScope": EnvVarScope.LOCAL in active_scopes,
+            },
+            "secretName": env_name,
+        },
+    )["secretsForScopes"]["secrets"]
+
+    for existing_secret in existing_secrets:
+        if len(existing_secret["locationNames"]) > 1:
+            raise click.ClickException(
+                f"Environment variable {env_name} is configured for multiple locations {', '.join(existing_secret['locationNames'])}, and cannot be modified via the CLI."
+            )
+
+    should_confirm = False
+    any_secret_is_global = any(
+        _secret_is_global(existing_secret) for existing_secret in existing_secrets
+    )
+    if not global_ and any_secret_is_global:
+        click.echo(
+            f"Environment variable {env_name} is set globally within the current deployment. The current command will only update the value for location {dg_context.project_name}. Use --global to update this value for all locations."
+        )
+        should_confirm = True
+
+    existing_scopes = set(
+        itertools.chain.from_iterable(
+            _get_secret_scopes(existing_secret)
+            for existing_secret in existing_secrets
+            if global_ or existing_secret["locationNames"]
+        )
+    )
+    if existing_scopes:
+        click.echo(
+            f"Environment variable {env_name} is already configured for {', '.join(s.value for s in existing_scopes)} scope{location_suffix}."
+        )
+        should_confirm = True
```

Comment:
wondering if this should always ask for confirmation in every case (unless you pass in -y or something)? At least if its affecting prod? Or is that too paternalistic? Can see it going either way. I do think we probably want an arg that automatically skips confirmation though in case people are setting this up programatically?

Reply:
Added the way to override confirmation